rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -2887,19 +2895,30 @@ public boolean isConnected(@HeaderParam("sessionid")
     @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
     @Path("login")
     @Produces("application/json")
-    public String login(@FormParam("username") String username, @FormParam("password") String password)
-            throws LoginException, SchedulerRestException {
+    public String login(@FormParam("username") String username, @FormParam("password") String password) {
         try {
-            if ((username == null) || (password == null)) {
+            if ((username == null || username == "") || (password == null || password == "")) {
```

Comment:
Generally it might be not super safe to compare string directly, e.g.:
`String a = "GitHub is awesome".substring(6, 6);`
then a == "" returns false.

so you can do:
`username == null || username.trim().empty()`

Reply:
fixed